src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.html

```diff
@@ -118,12 +120,17 @@ <h4 jhiTranslate="artemisApp.fileUploadAssessment.dashboard.heading" [translateV
                                         </span>
                                     </a>
                                     <!-- button for existing result with lock-->
-                                    <span *ngIf="submission.results && !submission?.results[correctionRound].completionDate">
+                                    <span
+                                        *ngIf="
+                                            submission.results &&
+                                            !(submission?.results && submission.results[correctionRound] && submission.results[correctionRound].completionDate)
```

Comment:
`submission?.results` is not needed because of the first condition. (If (`!submission?.results` is true, the first condition was false and vice versa.)

Reply:
I removed this part